anndata/tests/test_layers.py

```diff
@@ -62,6 +62,20 @@ def test_set_dataframe(homogenous, df, dtype):
     assert np.issubdtype(adata.layers["df"].dtype, dtype)
 
 
+def test_pickle():
+    import pickle
+    from anndata.core.aligned_mapping import Layers
+
+    adata = AnnData(X=X, layers=dict(L=L.copy()))
+
+    pickle_data = pickle.dumps(adata.layers)
+    layers: Layers = pickle.loads(pickle_data)
+    assert np.allclose(layers.parent.X, adata.X)
+    assert layers.keys() == adata.layers.keys()
+    for new, old in zip(layers.values(), adata.layers.values()):
+        assert np.allclose(new, old)
```

Comment:
Hmm. Any reason this isn't tested for the other types? Or for views?

And that `np.allclose` should be exact equality.

Reply:
It’s exemplary. If one works, the others do too. It’s to see if `__getnewargs_ex__` works.

Reply:
Yeah, but I was pretty surprised by how many things failed when I started parameterizing tests by sparse matrix type. Previously anndata (pre-6.21 I think) did not do well with CSC matrices.

Just for safety, I think it's worth chucking a fixture on this.